### ISSUE TITLE
Introduce students to the roles of QA and scrum master

### ISSUE THREAD

**Issue by shiryz** [foundersandcoders/master-reference#603]
For Nazareth's week 8 of express we have introduced the students to 2 of the agile roles as an introduction to how it's going to work in the second half of the course and because they lacked proper work and git flow.

The roles:
- Scrum master
  - Was present during planning.
  - made sure the students were on the same page.
  - helped scope the project.
  - helped identify potential blocks and problems that will come up during development.
  - made sure issues were raised before starting to work.
  - had standups with the team.
  - watched the repo and checked issues being raised to make sure the issues had sufficient description.

- QA (a better title would be process manager)
  - Merged prs.
  - made sure every pr has a description.
  - checked if commits relate to issues.
  - made sure the other team members have approved the pr with a comment.
    - if the pr has a comment/ changes requested then the qa can merge it because the other members have looked through the code.
    - if the pr doesn't have any comments and has been approved then the qa should look through the code and make sure it wasn't approved blindly.

These roles weren't introduced fully since it is only a 2-day project and should be more self-led, but the way that it has been introduced now has proven to help with their planning and ensured the team is using git flow properly.
I don't have an answer on whether this should be done every week but I do think introducing it in week 1 would be better than week 8.
you can find the minutes from the mentors retro of week 8 [here](https://github.com/foundersandcoders/master-reference/blob/2138e6b4fe3daa4981e4d79665442d28adeae989/curriculum-planning/minutes/express/summer-2017-mentor-retrospective.md#projects) and that week's SGC [here](https://github.com/foundersandcoders/nazareth-programme/blob/master/stop-go-continue/facn2/sgc-notes-week8.md#qascrum-master)

**Comment by jsms90** [comment]
Introducing this in week 1 might solve a lot of problems that don't currently get addressed in the first 8 weeks: #405 #418 #402 :+1: 

For FAC11, we have @Cleop giving a talk on Agile & the Scrum Master role #582, then introducing workflow on the first day of the build sprint #581. I've spoken to some alumni in FAC London and so far it feels like we don't have the resources to provide a Scrum Master & process manager for each team.

@shiryz are you planning on keeping these roles for FACN2's internal project?

@eliascodes what are your plans for Gaza?

**Comment by shiryz** [comment]
@jsms90 yes for sure, I think it's crucial to have these roles in such projects (longer than week).

**Comment by eliascodes** [comment]
cc @mantagen

**Comment by jsms90** [comment]
In London, it's not just that we don't have the resources. I'm actually not sure whether such active involvement from alumni is really preferable. But I'm really interested to see how the two approaches go in the different locations.

In London, we will be playing more of a hands off role.

**Pros:**
+ Agile teams should be self-organising. Having an alumnus acting as a Scrum Master that's external to the team can lead to the feeling that stand-ups are more about reporting to the alumni.
+ Greater sense of autonomy & ownership over their projects was greatly enjoyed by FAC10

**Cons:**
+ Being more hands off means that FAC cannot be as intimately aware of whether the students are achieving the learning objectives
+ Less opportunity to introduce expertise from the alumni network

At the moment, I'm leaning more towards the idea that alumni should perform a facilitation role when it comes to these projects, rather than actually occupying roles on the teams. The thing that students would undeniably benefit from is assistance during project scoping. Recent graduates frequently suffer from poor scoping due to inexperience, which leads to projects overrunning & they end up working for many weeks for free.

Where is "scoping" support relevant:
+ Week 10 ideation phase - @nogainbar and I were floating today
+ Week 11-12 - @Cleop and I will be floating during FAC11's sprint planning meetings

At the moment, I'm planning on staggering FAC11's daily stand-ups (weeks 11-12) and being present at each of them, for at least the first few days. The idea is to identify any areas where the existing expertise within our alumni base can be tapped up. For example, if a group is working with D3 for the first time, I can point them to alumni who have done a grad project / post-grad project with D3.

**Comment by mantagen** [comment]
@jsms90 what is London thinking about the PA roles and their respective allocations?
@shiryz what is Nazareth thinking about the PA roles and their respective allocations?
